tests/robustness/main_test.go

```diff
@@ -114,8 +143,8 @@ func (s testScenario) run(ctx context.Context, t *testing.T, lg *zap.Logger, clu
 		defer close(finishTraffic)
 		err := failpoint.Inject(ctx, t, lg, clus, s.failpoint)
 		if err != nil {
-			t.Error(err)
 			cancel()
+			t.Fatal(err)
```

Comment:
Just trying to understand this better, if we use `t.Fatal` here, we won't end up retrying the test right?
Referring to the comment here: https://github.com/etcd-io/etcd/pull/17725#issuecomment-2041203978